fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreCountRecordsTest.java

```diff
@@ -651,6 +649,372 @@ void countKeyOnNewStore() throws ExecutionException, InterruptedException {
         }
     }
 
+    static Stream<Arguments> disableRecordCountKey() {
+        return Stream.of(true, false)
+                .flatMap(fromWriteOnly -> Stream.of(true, false)
+                        .flatMap(hasFallBack -> Stream.of(true, false)
+                                .map(grouped -> Arguments.of(fromWriteOnly, hasFallBack, grouped))));
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    void disableRecordCountKey(boolean fromWriteOnly, boolean hasFallback, boolean grouped) {
+        final RecordMetaDataBuilder metaDataBuilder = simpleMetaDataBuilder();
+        addRecordCountKey(metaDataBuilder, grouped ? GROUP_EXPRESSION : EmptyKeyExpression.EMPTY);
+        if (hasFallback) {
+            addCountIndex(metaDataBuilder, grouped ? GROUP_EXPRESSION : EmptyKeyExpression.EMPTY);
+        }
+        IntConsumer assertCount = expected -> {
+            if (grouped) {
+                for (int groupingValue = 1; groupingValue < 6; groupingValue++) {
+                    assertGroupedCount(expected / 5, 1);
```

Comment:
```suggestion
                    assertGroupedCount(expected / 5, groupingValue);
```

I think this is what you meant here?

Reply:
Yes that is what I meant